### ISSUE TITLE
AZ LOGIN failing

### ISSUE THREAD

**Issue by centipod** [Azure/azure-cli#19595]
## Describe the bug

**Command Name**
`az login`

**Errors:**
```
local variable 'token_entry' referenced before assignment
Traceback (most recent call last):
python3.8/site-packages/knack/cli.py, ln 215, in invoke
    cmd_result = self.invocation.execute(args)
cli/core/commands/__init__.py, ln 654, in execute
    raise ex
cli/core/commands/__init__.py, ln 717, in _run_jobs_serially
    results.append(self._run_job(expanded_arg, cmd_copy))
cli/core/commands/__init__.py, ln 710, in _run_job
    six.reraise(*sys.exc_info())
...
cli/command_modules/profile/custom.py, ln 150, in login
    subscriptions = profile.find_subscriptions_on_login(
azure/cli/core/_profile.py, ln 212, in find_subscriptions_on_login
    subscriptions = subscription_finder.find_from_user_account(
azure/cli/core/_profile.py, ln 851, in find_from_user_account
    if not token_entry:
UnboundLocalError: local variable 'token_entry' referenced before assignment
```

## To Reproduce:
Steps to reproduce the behavior. Note that argument values have been redacted, as they may contain sensitive information.

- _Put any pre-requisite steps here..._
- `az login --user {} --tenant {}`

## Environment Summary
```
macOS-11.5.2-x86_64-i386-64bit
Python 3.8.6
Installer: HOMEBREW

azure-cli 2.15.0 *

```

**Comment by yonzhan** [comment]
@jiasli for awareness

**Comment by jiasli** [comment]
This issue has been fixed by https://github.com/Azure/azure-cli/pull/15962 and released in 2.15.1.

Please update to the latest version.
